nmpolicy/internal/resolver/errors.go

```diff
@@ -16,24 +16,45 @@
 
 package resolver
 
-import "fmt"
+import (
+	"fmt"
 
-func pathError(format string, a ...interface{}) error {
-	return fmt.Errorf("invalid path: %v", fmt.Errorf(format, a...))
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/ast"
+)
+
+type PathError struct {
+	inner     error
+	errorNode ast.Node
```

Comment:
This should be a pointer.

Reply:
why?

Reply:
To detect if we are missing setting it (by a bug or the like) is better if we have a pointer if we don't it's more difficult.

Reply:
done